regression-test/suites/nereids_rules_p0/mv/partition_mv_rewrite.groovy

```diff
@@ -371,4 +368,165 @@ suite("partition_mv_rewrite") {
     order_qt_query_16_0_after "${ttl_partition_sql}"
 
     sql """ DROP MATERIALIZED VIEW IF EXISTS ${ttl_mv_name}"""
+
+
+    // date roll up mv
+    def roll_up_mv_def_sql = """
+    select date_trunc(`l_shipdate`, 'day') as col1, l_shipdate, o_orderdate, l_partkey,
+    l_suppkey, sum(o_totalprice) as sum_total
+    from lineitem
+    left join orders on lineitem.l_orderkey = orders.o_orderkey and l_shipdate = o_orderdate
+    group by
+    col1,
+    l_shipdate,
+    o_orderdate,
+    l_partkey,
+    l_suppkey;
+    """
+
+    def roll_up_all_partition_sql = """
+    select date_trunc(`l_shipdate`, 'day') as col1, l_shipdate, o_orderdate, l_partkey,
+    l_suppkey, sum(o_totalprice) as sum_total
+    from lineitem
+    left join orders on lineitem.l_orderkey = orders.o_orderkey and l_shipdate = o_orderdate
+    group by
+    col1,
+    l_shipdate,
+    o_orderdate,
+    l_partkey,
+    l_suppkey;
+   """
+
+    def roll_up_partition_sql = """
+    select date_trunc(`l_shipdate`, 'day') as col1, l_shipdate, o_orderdate, l_partkey,
+    l_suppkey, sum(o_totalprice) as sum_total
+    from lineitem
+    left join orders on lineitem.l_orderkey = orders.o_orderkey and l_shipdate = o_orderdate
+    where (l_shipdate>= '2023-10-18' and l_shipdate <= '2023-10-19')
+    group by
+    col1,
+    l_shipdate,
+    o_orderdate,
+    l_partkey,
+    l_suppkey;
+    """
+
+    sql """DROP MATERIALIZED VIEW IF EXISTS ${mv_name}"""
+    sql """DROP TABLE IF EXISTS ${mv_name}"""
+    sql"""
+        CREATE MATERIALIZED VIEW ${mv_name}
+        BUILD IMMEDIATE REFRESH AUTO ON MANUAL
+        partition by (date_trunc(`col1`, 'month'))
+        DISTRIBUTED BY RANDOM BUCKETS 2
+        PROPERTIES ('replication_num' = '1')
+        AS
+        ${roll_up_mv_def_sql}
+        """
+    waitingMTMVTaskFinished(getJobName(db, mv_name))
+
+    explain {
+        sql("${roll_up_all_partition_sql}")
+        contains("${mv_name}(${mv_name})")
```

Comment:
why 2 ${mv_name}？

Reply:
because the key word in explain plan is  `mv_name(mv_name)` when use the materialized view